source/Calamari.Azure/Kubernetes/Discovery/AzureKubernetesDiscoverer.cs

```diff
@@ -14,7 +14,7 @@ public AzureKubernetesDiscoverer(ILog log) : base(log)
         {
         }
 
-        public override string Name => KubernetesAuthenticationContextTypes.AzureServicePrincipal;
```

Comment:
Can we reduce the chance of a similar problem in future by defining these  in constant strings that get referenced, instead of using string literals in multiple places? Might be hard because of the Calamari/Server split?
Whatever we end up doing here, we could also re-visit it when Calamari is all back in the server solution.

Reply:
At this stage I don't think we have any shared projects across server and Calamari so I think the best option is to revisit this when calamari is merged into the Server Solution.

Reply:
Ok. As per our conversation, let's just get some comments here and for the equivalent `"Aws"` type, indicating which server code they need to keep in sync with, and noting that we can revisit to move to shared code after the mono-repo changes are done.